src/core/public/plugins/plugin.ts

```diff
@@ -115,6 +115,9 @@ export class PluginWrapper<
     }
 
     this.instance = undefined;
+
+    // some plugins might be awaiting for core.getStartServices(); reject these promises
+    this.startDependencies$.complete();
```

Comment:
1. We probably want a unit test to assert that change of behavior
2. We also may want to do that in the server-side counterpart of the pluginWrapper:

https://github.com/elastic/kibana/blob/6cb2898d2eefd1e3bb77bc54a40be5f5d4e8e6aa/src/core/server/plugins/plugin.ts#L140-L143